Add release process documentation

michaelstingl opened the pull request:
Fixes #48

## Changes
This PR adds a minimal release process section to CONTRIBUTING.md as discussed in #48.

### What's included:
- **Release authority**: Clear statement that only maintainers can create releases
- **Release process**: Simple 4-step checklist for creating releases
- **Version guidelines**: Clear conventions for 0.x.x versioning:
  - `0.x.0` for breaking changes
  - `0.x.y` for new features (backwards compatible)
  - `0.x.y-z` for bug fixes only

This complements the version stability disclaimer from PR #43 and provides the structure @butonic highlighted we need for managing releases.

The approach is intentionally minimal and lightweight to fit our community-driven model.

butonic commented on .github/workflows/publish-helm-charts.yml:
```diff
@@ -22,6 +22,44 @@ jobs:
+          
+          # Basic semver check (x.y.z format)
+          for version in "$PROD_VERSION" "$DEV_VERSION"; do
+            if ! [[ "$version" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
```
this will not match `0.x.y-z` - Bug fixes only

michaelstingl commented:
@butonic Thanks for catching the pre-release version issue\! Fixed in 45a8ad1.

However, while implementing this, I discovered a much more serious problem: **The workflow was publishing charts on EVERY push to main**, potentially overwriting the same version multiple times. This violates the fundamental principle that Helm chart versions should be immutable.

I've addressed this with additional commits:
- fcc1c7e - Only publish on tag pushes (removed main branch trigger)
- cf1e9aa - Added test build support for manual triggers (separate namespace)

Now:
- ✅ Official releases: Only on tag push to `helm-charts/`
- ✅ Test builds: Manual trigger to `helm-charts-test/` with timestamp versions
- ✅ Pre-release versions like `0.1.2-1` are supported

This should prevent accidental version overwrites and follow Helm best practices.
